control-plane/p4RuntimeSerializer.cpp

```diff
@@ -197,6 +197,7 @@ struct MatchField {
     const uint32_t bitwidth;  // How wide this field is.
     const IR::IAnnotated* annotations;  // If non-null, any annotations applied
                                         // to this field.
+    const cstring type_name;  // The Type_Newtype expression name of this field.
```

Comment:
Be more precise in the comment. I imagine that the string is empty if the type of the match field expression is not a user-defined type.